src/components/tabs/js/tabsController.js

```diff
@@ -369,6 +369,11 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
     if (destroyed) return;
     var selectedIndex = ctrl.selectedIndex,
         tab           = ctrl.tabs.splice(tabData.getIndex(), 1)[ 0 ];
+    angular.forEach(ctrl.tabs, function(remainingTab) {
+      if(remainingTab.index > tab.index) {
```

Comment:
Didn't take a look at the logic behind your PR yet.

Nitpicky: But the `if` should have space between the expression. Just for consistency.